torch/jit/_script.py

```diff
@@ -961,9 +960,11 @@ def forward(self, input):
             )
         if _rcb is None:
             _rcb = _jit_internal.createResolutionCallbackFromFrame(_frames_up + 1)
+        qualified_name = _qualified_name(obj)
```

Comment:
What happens if we pass a TorchScript builtin class name, an interface class, or an Enum class to `torch.jit.script`? Some of them may be detected by the "not-support-inheritance" checking, but interface class may not be (please double check). But even for those that are currently covered by "not-support-inheritance", if we support inheritance in the future, this could be a potential bug. So it is better to explicitly check against those cases (I am not sure if there is a flag to know if a Python class is a built-in TorchScript class).

Reply:
I see, if we add a longer list of items to check here, users would get better clue on what they did wrong, I will add them.

To clarify, what do you mean by "TorchScript builtin class"?

Reply:
TorchScript built-in class refers to any Python class that is recognized as built-in types in Torchscript. Those classes were not compiled into TorchScript IR, but directly internalized in TorchScript. I don't know what would be the behavior for something like `torch.jit.script(torch.Tensor)` ....

Here is the list I could think of:
- namedtuple
- TSBuiltinClass := TSTensor | "torch.device" | "torch.Stream" | "torch.dtype" | 
                  "torch.nn.ModuleList" | "torch.nn.ModuleDict" | ... (do we have a complete list here?)
TSTensor := "torch.Tensor" | "common.SubTensor" | "common.SubWithTorchFunction" | 
            "torch.nn.parameter.Parameter" | and subclasses of torch.Tensor

- `torch.nn.Module`

This is probably not a complete list. If we have a is_torchscript_built_in flag for a TorchScript (Python) class, it probably would be a more sustainable solution (because the list of built-in classes will likely grow).